### PR TITLE
Update page layout components for the new theme

### PR DESCRIPTION
**Pull request by CarlosCortizasCT**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=Iel7A-m)

#### Summary

This is a follow-up work from #2870 where we refactor the components for the new theme where design tokens wasn't enough.

#### Description

These are the key point that are missing:
- [x] Page subtitles should have a different color
- [x] Info main page should not have a divider between header and content
- [x] Dialogs main paddings should be bigger
- [x]  Dialogs titles should use H3 tags
- [x] Modals should use H1 tag for titles (when using `big` for `titleSize`)
- [x] Modals close icon should be big
- [x] Tab controls labels should use H3 tags (instead of H4)
- [x] ~~Form controls buttons should be in the same _row_ as title/subtitle~~ (to be addressed in #2813)

### CODE REVIEW

File: packages/application-components/src/components/dialogs/internals/dialog-container.tsx [hunk 1]
```diff
@@ -109,6 +110,7 @@ const DialogContainer = (props: Props) => (
             css={css`
               min-height: 0;
               ${sizeStyles(props)}
+              padding: ${designTokens.paddingForDialogContainer};
```
Comment (CarlosCortizasCT) [inline packages/application-components/src/components/dialogs/internals/dialog-container.tsx:113]: Dialogs are using the `Card` component which internally uses the `Spacing.Inquish` component. The latter does not have a scale which fits with the expected paddings for Dialogs so we use a custom padding here to add what it's missing from the Card paddings to meet the desired values.

**Review by emmenko — APPROVED** [review]
Nice! 🥳

**Comment by emmenko** [comment]
> Form controls buttons should be in the same row as title/subtitle

FYI: This should be a separate refactoring, as we need better UI requirements first (see #2813)

File: packages/application-components/src/components/main-pages/custom-form-main-page/custom-form-main-page.tsx [hunk 2]
```diff
@@ -42,10 +42,19 @@ type CustomFormMainPageProps = {
    * Determines if the form controls should be rendered.
    */
   hideControls?: boolean;
+  /**
+   * Determines whether the divider between header and content should be rendered.
+   * (default: true)
```
[the remark this answers is not in the document]
Comment (CarlosCortizasCT) [reply packages/application-components/src/components/main-pages/custom-form-main-page/custom-form-main-page.tsx]: Updated in this commit: d4b07fa3fdc072df5214e4a4fcd32f3c7c9f81f5

**Comment by CarlosCortizasCT** [comment]
Percy failing is because of the new icons in `ui-kit`.

**Review by emmenko — APPROVED** [review]
💯

**Review by kark — APPROVED** [review]
Looking good 🎉

**Review by Rhotimee — APPROVED** [review]
🚀

**Comment by emmenko** [comment]
Got the UX approval, will merge it tomorrow

**Pull request merged**